### ISSUE TITLE
Deleting with invalid indexes give different and (some wrong) error messages

### ISSUE THREAD

**Issue by ekweirui** [ekweirui/pe#6]
![image.png](https://raw.githubusercontent.com/ekweirui/pe/main/files/b112a4f9-b986-4754-9ce9-cb817fc5d2bb.png)

![image.png](https://raw.githubusercontent.com/ekweirui/pe/main/files/1d648f5d-5512-4409-9800-3358eae8b73c.png)

In the first case, I delete 4, which is an invalid index, and that gave me an invalid index error message. 
In the second case I delete 0, and that gave me an invalid command format error.

The second case should not occur as this is not a case of invalid command format, but more of an invalid index error.
The first case was handled correctly, but somehow got left out for the second case.

**Comment by nus-pe-script** [comment]
# Team's Response
We do not think that this is a bug as the error message is valid. “Delete 4” returns the guest index provided as invalid as it is correct that a positive integer was provided, but simply out of the range of the length of the number of guests. However, delete 0 is an invalid command format, as we did specify that the INDEX provided has to be positive. Thus, the command format provided was invalid.


![image.png](https://raw.githubusercontent.com/nus-cs2103-AY2223S1/pe/master/files/92a5f725-93aa-4039-8fcf-8dfd3a6b74fd.png)
# Items for the Tester to Verify
## :question: Issue response

Team chose [`response.Rejected`]

- [x] I disagree

**Reason for disagreement:** I feel that the bug boils down to the same root, which is that the user inputted an invalid index. From the view point of the end user, there is no difference in the command format as both are typing the exact same words ie `delete ...`, and the only differing thing is in the index itself. Thus, it would feel weird for users to see different error messages for a bug that has the same root.  

In addition, if the usage of the command format is used to validate positive indexes, then I feel that the error message of the "guest index provided is invalid" should have been modified and improved to something along the line of "index is too large/index is more than the number of guests". This is because you have already "handled" the use case whereby it's not positive in the command format validation.

However, this was not the case, which tells me that the developers have never intended for it to be this way.  
And besides, this way is also not ideal as the same type of error is being handled differently, which is not ideal as it reduces cohesion.

In conclusion, I feel that the developers should have 
1) either handled the same type of error in one place for consistency's sake  
2) or handled the error in different places but give specific error messages since the developers can now afford to do so

-------------------
